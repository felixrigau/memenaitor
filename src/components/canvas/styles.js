const styles = {
    canvas: {
        width: '100%',
        height: '100%',
        border: '1px dashed red',
        position: 'relative',
    },
};

export default styles;
