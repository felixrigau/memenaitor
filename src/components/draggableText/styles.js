const styles = {
    text: props => ({
        position: 'absolute',
        margin: 0,
        textShadow: `${props.text.shadowColor} -1px -1px 0px, ${props.text.shadowColor} 1px -1px 0px, ${props.text.shadowColor} -1px 1px 0px, ${props.text.shadowColor} 1px 1px 0px`,
        fontFamily: 'Impact'
    }),
}

export default styles