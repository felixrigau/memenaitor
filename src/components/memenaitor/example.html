<div class="Canvas_canvas__1fiR0" style="width: 291.8px; height: 291.8px; background-color: rgb(255, 255, 255);">
    <div class="Canvas_canvasClickCollector__2p1Cq"></div>
    <div class="RndWrapper_rndLayer__2hJkb RndWrapper_selectedLayer__3_bVu react-draggable react-draggable-dragged" style="position: absolute; user-select: auto; touch-action: none; width: 92.6848%; height: 24.7032%; display: inline-block; top: 0px; left: 0px; cursor: move; transform: translate(10.9px, 11.3106px); max-width: 9.0072e+15px; max-height: 9.0072e+15px; box-sizing: border-box;">
        <div class="Text_text__2Pwwm" style="text-align: center; font-family: &quot;Permanent Marker&quot;; color: rgb(255, 23, 68); font-weight: bold; text-shadow: black -1px -1px 0px, black 1px -1px 0px, black -1px 1px 0px, black 1px 1px 0px; background: rgba(0, 0, 0, 0.05);">
            <span class="textFitted" style="display: inline-block; font-size: 44px;">Sample text</span>
        </div>
        <div class="Text_editTextButton__3R_Hp">
            <span class="Text_editText__3KfER">Edit</span>
            <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" height="1em" width="1em" style="vertical-align: middle;">
                <path d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"></path>
            </svg>
        </div>
        <div class="DuplicateLayer_button__X8Ykf">
            <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" height="1em" width="1em" style="vertical-align: middle;">
                <path d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"></path>
            </svg>
        </div>
        <div class="DeleteLayer_button__3PJcL">
            <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 448 512" height="1em" width="1em" style="vertical-align: middle;">
                <path d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"></path>
            </svg>
        </div>
        <span>
            <div style="position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;">
            </div>
            <div style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"></div>
            <div style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"></div>
            <div style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;"></div>
            <div class="RndWrapper_resizeHandle__3B1w7" style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; top: -10px; cursor: nw-resize;"></div>
            <div class="RndWrapper_resizeHandle__3B1w7" style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; top: -10px; cursor: ne-resize;"></div>
            <div class="RndWrapper_resizeHandle__3B1w7" style="position: absolute; user-select: none; width: 20px; height: 20px; left: -10px; bottom: -10px; cursor: sw-resize;"></div>
            <div class="RndWrapper_resizeHandle__3B1w7" style="position: absolute; user-select: none; width: 20px; height: 20px; right: -10px; bottom: -10px; cursor: se-resize;"></div>
        </span></div><div class="__resizable_base__" style="width: 100%; height: 100%; position: absolute; transform: scale(0, 0); left: -2.14748e+09px; flex: 0 1 0%;">
    </div>
</div>