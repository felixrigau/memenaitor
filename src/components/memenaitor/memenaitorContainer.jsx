import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import Memenaitor from './memenaitor';

export class MemenaitorContainer extends Component {
    render() {
        return (
            <Fragment>
                <Memenaitor {...this.props}/>
            </Fragment>
        );
    }
}

const mapStateToProps = state => ({
    selectedElement: state.selectedElement,
});

const mapDispatchToProps = {};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MemenaitorContainer);
