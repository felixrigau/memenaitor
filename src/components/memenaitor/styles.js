const styles = {
    container: {
        width: props => `${props.width.toString()}px`,
        height: props => `${props.height.toString()}px`,
        backgroundColor: '#f7d13d',
        margin: '100px 0 0 100px'
    },
    resultContainer:{
        position: 'relative',
        margin: '1px 0 0 0'
    },
    logo:{
        position: 'absolute'
    },
    canvas:{
        position: 'absolute'
    },
    box: {
        width: '50px',
        height: '50px',
        backgroundColor: '#ebebeb',
        border: '2px solid #ababab',
        margin: '50px 0 0 50px'
    },
    svg: {
        position: 'absolute',
        border: '2px solid red',
    },
    textSvg: {
        cursor: 'pointer',
        '&::selection': { 
            color: 'transparent'
        }
    }
};

export default styles;
