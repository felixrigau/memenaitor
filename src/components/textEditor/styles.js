const styles = {
    textEditor: {
        display: 'flex',
        flexDirection: 'column'
    }
};

export default styles;
