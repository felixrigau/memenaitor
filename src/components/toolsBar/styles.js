const styles = {
    toolsBar: {
        border: '1px solid black',
        width: '100%',
        height: '50px',
    },
};

export default styles;
