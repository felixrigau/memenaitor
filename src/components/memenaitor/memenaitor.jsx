import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import styles from './styles';
import vue from './../../assets/vue.jpg';

export class Memenaitor extends Component {
    constructor(props) {
        super(props);
        this.state = {
            texts: {
                top: {
                    text: '',
                    location: {
                        x: 0,
                        y: 0,
                    },
                },
                bottom: {
                    text: '',
                    location: {
                        x: 0,
                        y: 0,
                    },
                },
            },
            image: '',
        };
    }

    componentDidMount() {
        // this.createSVG();
        const textRed = document.querySelector('#textRed');
        textRed.addEventListener('mousemove', e => this.handleMouseMove(e))
        const textBlue = document.querySelector('#textBlue');
        textBlue.addEventListener('mousemove', e => this.handleMouseMove(e))
    }

    handleClick = e => {
        debugger
        console.log('offsetX', e.offsetX)
        console.log('offsetX', e.offsetX)
    }

    handleMouseMove = e => {
        if (e.buttons === 1) {
            // debugger
            const element = e.target;       
            if (element.id.includes('text')) {
                this.changePosition(e.target, 'x', e.movementX)
                this.changePosition(e.target, 'y', e.movementY)
            }
        }
        
    }

    changePosition = (element, coordinate, value) => {
        // debugger
        const position = parseInt(element.getAttributeNS(null, coordinate));
        element.setAttributeNS(null, coordinate, position + value)
    }

    updateText = (position, text, x, y) => {
        this.setState({
            [position]: {
                text: text,
                location: {
                    x: x,
                    y: y,
                },
            },
        });
    };

    createSVG = () => {
        // const xmlns = 'http://www.w3.org/2000/svg';
        // let svgElem = document.createElementNS(xmlns, 'svg');
        // let txtElem1 = document.createElementNS(xmlns, 'text');
        // let txtElem2 = document.createElementNS(xmlns, 'text');

        // txtElem1.setAttributeNS(null, 'x', 100);
        // txtElem1.setAttributeNS(null, 'y', 100);
        // txtElem1.setAttributeNS(null, 'font-size', 28);

        // txtElem2.setAttributeNS(null, 'x', 100);
        // txtElem2.setAttributeNS(null, 'y', 250);
        // txtElem2.setAttributeNS(null, 'font-size', 28);

        // const helloTxt1 = document.createTextNode('Hello World!');
        // const helloTxt2 = document.createTextNode('Hello World!');

        // txtElem1.appendChild(helloTxt1);
        // txtElem2.appendChild(helloTxt2);

        // svgElem.appendChild(txtElem1);
        // svgElem.appendChild(txtElem2);

        const svg = document.querySelector('#svg');
        let logo = document.getElementById('logo');
        debugger
        this.convertSVGtoCanvas(svg, logo);
        // let element = document.getElementById('test');
        // element.appendChild(svgElem);
        // document.documentElement.appendChild(txtElem1);
    };

    convertSVGtoCanvas = (svgElem, logo) => {
        let result = document.getElementById('result');
        const svg_xml = new XMLSerializer().serializeToString(svgElem);
        let canvas = document.getElementById('canvas');
        canvas.width = result.width;
        canvas.height = result.height;
        var ctx1 = canvas.getContext('2d');

        var img = new Image();
        img.src = 'data:image/svg+xml;base64,' + btoa(svg_xml);

        img.onload = function() {
            ctx1.drawImage(logo, 0, 0, 300, 300);
            ctx1.drawImage(img, 0, 0, 300, 300);
            const base64 = canvas.toDataURL();
            canvas.width = 0;
            canvas.height = 0;
            result.src = base64;
        };
    };

    render() {
        const { textTop, textBottom, width, height, classes } = this.props;
        return (
            <Fragment>
                {/* <div id="container" className={classes.container}>
                    <div id="box" className={classes.box}>

                    </div>
                </div> */}
                <div>
                    <input id="button" type="button" value="convert" onClick={this.createSVG}/>
                    <img id="result" className={classes.result} height={height} width={width}/>
                </div>
                <div id="resultContainer" className={classes.resultContainer}>
                    <img id="logo" className={classes.logo} src={vue} alt="logo" height={height} width={width}/>
                    <canvas id="canvas" className={classes.canvas} height="0" width="0" />
                    <svg id="svg" className={classes.svg} height={height} width={width}>
                        <text className={classes.textSvg} id="textRed" x="0" y="15" fill="red" fontSize="28">I love SVG!</text>
                        <text className={classes.textSvg} id="textBlue" x="0" y="50" fill="blue" fontSize="28">I love SVG!</text>
                    </svg>
                </div>
            </Fragment>
        );
    }
}

Memenaitor.propTypes = {
    classes: PropTypes.object.isRequired,
    textTop: PropTypes.string,
    textBottom: PropTypes.string,
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
};

Memenaitor.defaultProps = {
    textTop: 'Text top',
    textBottom: 'Text bottom',
    width: 300,
    height: 300,
};

export default injectSheet(styles)(Memenaitor);
