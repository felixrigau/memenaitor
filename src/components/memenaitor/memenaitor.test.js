import React from 'react';
import { Memenaitor } from './memenaitor';
import { shallow } from 'enzyme';
import transfromJssClasses from './../../utils/transfromJssClasses';
import styles from './styles';

describe('</Memenaitor>', () => {
    it('Should render the component correctly', () => {
        const wrapper = shallow(<Memenaitor classes={transfromJssClasses(styles)} />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });
});
