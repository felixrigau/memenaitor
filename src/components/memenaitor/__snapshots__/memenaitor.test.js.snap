// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Memenaitor> Should render the component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Memenaitor
    classes={
      Object {
        "container": "container",
      }
    }
    height={500}
    textBottom="Text bottom"
    textTop="Text top"
    width={500}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="container"
        >
          <p>
            Text top
          </p>
          <p>
            Text bottom
          </p>
        </div>,
        <input
          type="button"
          value="Mix"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Text top
            </p>,
            <p>
              Text bottom
            </p>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text top",
            },
            "ref": null,
            "rendered": "Text top",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text bottom",
            },
            "ref": null,
            "rendered": "Text bottom",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "button",
          "value": "Mix",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="container"
          >
            <p>
              Text top
            </p>
            <p>
              Text bottom
            </p>
          </div>,
          <input
            type="button"
            value="Mix"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Text top
              </p>,
              <p>
                Text bottom
              </p>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Text top",
              },
              "ref": null,
              "rendered": "Text top",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Text bottom",
              },
              "ref": null,
              "rendered": "Text bottom",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "button",
            "value": "Mix",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
